lib/vimeo.js

```diff
@@ -378,37 +378,47 @@ Vimeo.prototype.generateClientCredentials = function (scope, fn) {
  * @param  {string}   path      The path to the file you wish to upload
  * @param  {Function} callback  A function that is called when the upload is complete, or fails. 
  */
-Vimeo.prototype.streamingUpload = function (path, video_uri, callback) {
-    var _self = this;
-
-    if (!callback) {
-        callback = video_uri;
-        video_uri = undefined;
-    }
-
-    var options = {
-        method : video_uri ? 'PUT' : 'POST',
-        path : video_uri ? video_uri + '/files' : '/me/videos',
-        query : {
-            type : 'streaming'
-        }
-    };
-
-    this.request(options, function (err, ticket, status, headers) {
-        if (err) {
-            return callback(err);
-        }
-
-        var file = new FileStreamer(path, ticket.upload_link_secure);
-
-        file.ready(function () {
-            _self.request({
-                method : 'DELETE',
-                path : ticket.complete_uri
-            }, callback);
-        });
-
-        file.error(callback);
-        file.upload();
-    });
+Vimeo.prototype.streamingUpload = function (path, video_uri, callback, progress_callback) {
+	var _self = this;
+
+	if (!callback) {
+		callback = video_uri;
+		video_uri = undefined;
+	}
+
+	var options = {
+		method : video_uri ? 'PUT' : 'POST',
+		path : video_uri ? video_uri + '/files' : '/me/videos',
+		query : {
+			type : 'streaming'
+		}
+	};
+
+	this.request(options, function (err, ticket, status, headers) {
+		if (err) {
+			return callback(err);
+		}
+
+		var file = new FileStreamer(path, ticket.upload_link_secure);
+		
+		if (progress_callback) {
+			var check_progress = setInterval(function () {
```

Comment:
Do we have to do set interval here? This seems so weird. Can we just pass the progress callback into the filestreamer so we have it called exactly when it's updated, as opposed to periodically checking via the event handler?